### PR TITLE
GEOMESA-2393 Kafka - Support avro serialization

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Supports registering avro scheme with Confluent SchemaRegistry

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
- [ ] TODO make tryDeserializeVersions try whichever one worked the last time, instead of always kryo then avro

**Comment by elahrvivaz** [comment]
Got it tested and working against a confluent registry.

**Comment by elahrvivaz** [comment]
- [ ] TODO remove the confluent bits, the registration works but confluent expects it's own avro serialization format and ours doesn't work

File: geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/RecordVersions.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        logger.warn("This version of Kafka doesn't support message headers, serialization may be slower")
+        (_, _, _) => Unit
+    }
+  }
+
+  private val _getHeaders: ConsumerRecord[_, _] => Map[String, Array[Byte]] = {
+    consumerMethods.find(m => m.getName == "headers" && m.getParameterCount == 0) match {
+      case Some(method) =>
+        record => {
+          record.headers.iterator.next.key
```
Comment (elahrvivaz) [inline geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/RecordVersions.scala]: I think this needs to be removed @ddseapy 
```suggestion
```

**Comment by dseapy** [comment]
@jahhulbert-ccri Any objections to having this merged?

**Pull request merged**